File: LeetCodeRecord/MethodFirst/448-FindDisappearedNumbers.java
package MethodFirst;

import java.util.ArrayList;
import java.util.List;

class FindDisappearedNumbers {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> res = new ArrayList<Integer>();
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[Math.abs(nums[i]) - 1] > 0) {
                nums[Math.abs(nums[i]) -1] *= -1;
            }
        }

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                res.add(i + 1);
            }
        }

        return res;
    }    
}

File: LeetCodeRecord/ACFirst-Golang/448-FindDisappearedNumbers.go
package main

import (
	"math"
)

func findDisappearedNumbers(nums []int) []int {
	for i := 0; i < len(nums); i++ {
		num := int(math.Abs(float64(nums[i])))
		nums[num-1] = -int(math.Abs(float64(nums[num-1])))
	}

	res := []int{}
	for i := 0; i < len(nums); i++ {
		if nums[i] > 0 {
			res = append(res, i+1)
		}
	}

	return res
}
